Library/ViewModels/CommentRepliesViewModel.swift

```diff
@@ -27,25 +39,63 @@ public final class CommentRepliesViewModel: CommentRepliesViewModelType,
   CommentRepliesViewModelInputs,
   CommentRepliesViewModelOutputs {
   public init() {
-    let rootComment = Signal.combineLatest(
-      self.commentProperty.signal.skipNil(),
+    let rootCommentProject = Signal.combineLatest(
+      self.commentProjectProperty.signal.skipNil(),
       self.viewDidLoadProperty.signal
     )
     .map(first)
 
+    let rootComment = rootCommentProject.map(first)
+
+    let project = rootCommentProject.map(second)
+
     self.loadCommentIntoDataSource = rootComment
+
+    let currentUser = self.viewDidLoadProperty.signal
+      .map { _ in AppEnvironment.current.currentUser }
+
+    let inputAreaBecomeFirstResponder = rootCommentProject
+      .map(third)
+      .takeWhen(self.viewDidAppearProperty.signal)
+
+    self.configureCommentComposerViewWithData = Signal
+      .combineLatest(
+        project,
+        currentUser.signal,
+        self.viewDidLoadProperty.signal,
+        inputAreaBecomeFirstResponder
+      )
+      .map { ($0.0, $0.1, $0.3) }
+      .map { project, currentUser, inputAreaBecomeFirstResponder in
+        let isBacker = userIsBackingProject(project)
+        let isCreatorOrCollaborator = !project.memberData.permissions.isEmpty && !isBacker
+        let canPostComment = isBacker || isCreatorOrCollaborator
+
+        guard let user = currentUser else {
+          return (nil, false, true, false)
```

Comment:
For readability and completeness let's make the return type of this closure `-> CommentComposerViewData` and include the arguments in the tuple, something like:

```swift
guard let user = currentUser else {
  return (
    avatarURL: nil,
    canPostComment: false,
    hidden: true,
    becomeFirstResponder: false
  )
}
```